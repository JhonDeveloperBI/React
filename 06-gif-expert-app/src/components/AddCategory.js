import { React, useState } from 'react'

export const AddCategory = () => {

   const [inputValue, setinputValue] = useState('initialState')
   
   const handleInputChange =(e) => {
    setinputValue(e.target.value)
   }
   
   const handleSubmit = (e) => {
       e.preventDefault(); // no se refresca la pagina
       console.log("submit done");
   }

  return (
    <form onSubmit={ handleSubmit }>
    {/*  <h1> { inputValue }</h1> */}
        <input 
            type = "text" 
            value={ inputValue }
            onChange= { handleInputChange }
        />
    </form>
  )
}
