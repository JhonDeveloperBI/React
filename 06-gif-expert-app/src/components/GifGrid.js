import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [Images, setImages] = useState([]) //always init

    //it is used to solve the rendering problem of the component
    useEffect(() =>{
        getGifs();

    },[]) // componentDidmount


    const getGifs = async() => {

            const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI( category ) }&limit=10&api_key=A8xMXqzieIHmtO3BjGLAtf1daSSDAv8K`;
          
            const resp = await fetch( url );
            const { data } = await resp.json();
        
            const gifs = data.map( img => {
                return {
                    id: img.id,
                    title: img.title,
                    url: img.images?.downsized_medium.url
                }
            })
       setImages(gifs);
        }

      //  getGifs();

  return (
    <div>
        <h3>{ category }</h3>
       {/*  <ol>
            {
             Images.map( ({ id, title}) =>  (
               <li key={id}>{title}</li> 
              ))
            }
        </ol> */}

         {
             Images.map( ( img ) =>  (
               <GifGridItem  
                key= { img.id } 
                { ...img } //operator express propiedad independiente
                /> 
              ))
            }

    </div>
  )
}
