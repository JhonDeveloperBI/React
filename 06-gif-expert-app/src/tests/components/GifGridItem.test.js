import { GifGridItem } from "../../components/GifGridItem"
import { shallow } from "enzyme";

describe('test in GifGridItem', () => { 

    test('should show <GifGridItem /> correct', () => { 
        
        const title = 'test';
        const url = 'https://www.google.com';

        const wrapper = shallow(<GifGridItem title={title}  url = {url} />)

        expect( wrapper).toMatchSnapshot(); // snapshot
     })
 })