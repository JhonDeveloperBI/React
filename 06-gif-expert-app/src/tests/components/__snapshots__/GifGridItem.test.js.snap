// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGridItem should show <GifGridItem /> correct 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="test"
    src="https://www.google.com"
  />
  <p>
     
    test
     
  </p>
</div>
`;
